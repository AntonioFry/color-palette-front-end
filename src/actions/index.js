export const throwError = error => ({
  type: 'THROW_ERROR',
  payload: {
    error
  }
});

export const isLoading = isLoading => ({
  type: 'IS_LOADING',
  payload: {
    isLoading
  }
});

export const search = searchWord => ({
  type: 'SEARCH',
  payload: {
    searchWord
  }
});

export const addProject = project => ({
  type: 'ADD_PROJECT',
  payload: {
    project
  }
});

export const addPalette = palette => ({
  type: 'ADD_PALETTE',
  payload: {
    palette
  }
});

export const addProjects = projects => ({
  type: 'ADD_PROJECTS',
  payload: {
    projects
  }
});

export const addPalettes = palettes => ({
  type: 'ADD_PALETTES',
  payload: {
    palettes 
  }
});

export const removeProject = id => ({
  type: 'REMOVE_PROJECT',
  payload: {
    id
  }
});

export const removePalette = id => ({
  type: 'REMOVE_PALETTE',
  payload: {
    id
  }
});