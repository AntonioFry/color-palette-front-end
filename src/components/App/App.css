body {
  font-family: 'Lato', sans-serif;
}

main {
  display: block;
  justify-content: center;
  margin: 2%;
}

h1 {
  font-size: 4em;
  color: maroon;
  text-align: center;
}

span {
  color:rgb(61, 134, 199);
}

.forms {
  display: inline-flex;
  width: 100%;
}

.orStyle {
  color: maroon;
  font-size: 4em;
  font-family: "Lato";
  margin: 5%;
}


