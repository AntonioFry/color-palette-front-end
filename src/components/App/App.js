import React, { Component } from 'react';
import { getProjects, getPalettes } from '../../Utils/API/apiCalls';
import { setProjects, setPalettes } from '../../actions';
import { connect } from 'react-redux';
import './App.css';
import { connect } from 'react-redux'
import { addProjects, addPalettes } from '../../actions';
import GeneratedColors from '../GeneratedColors/GeneratedColors';
import PaletteForm from '../PaletteForm/PaletteForm';
import ProjectContainer from '../ProjectContainer/ProjectContainer';
import ProjectForm from '../ProjectForm/ProjectForm';

export class App extends Component {
 async componentDidMount() {
    try {
      const projects = await getProjects();
      // this.props.setProjects(projects);
      const palettes = await getPalettes();
      this.props.addProjects(projects)
      this.props.addPalettes(palettes)
    } catch (error) {
      throw new Error(`failed to fetch: ${error.message}`)
    }
  }

  render() {
    return (
      <main>
        <header>
          <h1>Palette P<span>!</span>cker</h1>
        </header>
        <ProjectForm/>
<<<<<<< HEAD
        <PaletteForm/>
        <GeneratedColors/>
=======
        <GeneratedColors/>
        <PaletteForm/>
        <ProjectContainer/>
>>>>>>> 8ea6e7663783141335cf1612e6137758f822d00d
      </main>
    )
  }
}


const mapDispatchToProps = dispatch => ({
  addProjects: (projects => dispatch(addProjects(projects))),
  addPalettes: (palettes => dispatch(addPalettes(palettes)))
})

export default connect(null, mapDispatchToProps)(App)

