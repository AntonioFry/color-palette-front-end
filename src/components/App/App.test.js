import React from 'react';
import { App } from './App.js';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './App';
import { addProjects, addPalettes } from '../../actions';

describe('App', () => {

  describe('App component',  () => {
    let wrapper;
    let props;

    beforeEach(() => {
      props = {
        palettes: [{}, {}]
      }
      wrapper = shallow(<App palettes={props.palettes} />)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it.skip('should throw an error if componentDidMount cathces an error', () => {
      window.fetch = jest.fn().mockImplementation(() => {
        return Promise.resolve({
          ok: false
        })
      });
    
      expect(wrapper.instance().componentDidMount()).toEqual()
    })
  
  })

  describe('mapStateToProps', () => {
    it('should return array with all palettes', () => {
      const mockStore = {
        palettes: [
          { id: 1, palette_name: 'Dog Colors' },
          { id: 2, palette_name: 'Cat Colors' },
          { id: 3, palette_name: 'Something' }
        ]
      }

      const expected =  {
        palettes:  [
          { id: 1, palette_name: 'Dog Colors' },
          { id: 2, palette_name: 'Cat Colors' },
          { id: 3, palette_name: 'Something' }
        ] 
    }

      const mappedProps = mapStateToProps(mockStore);
      expect(mappedProps).toEqual(expected);
    });
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with addProjects', () => {
      
       
    });

    it('should call dispatch with addPalettes', () => {

    });
  })
})