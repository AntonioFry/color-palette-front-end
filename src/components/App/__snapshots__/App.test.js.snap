// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should match the snapshot 1`] = `
<main>
  <header>
    <h1>
      Palette P
      <span>
        !
      </span>
      cker
    </h1>
  </header>
  <Connect(GeneratedColors) />
  <section
    className="forms"
  >
    <Connect(ProjectForm) />
    <h3
      className="orStyle"
    >
      OR
    </h3>
    <Connect(PaletteForm) />
  </section>
  <Connect(ProjectContainer) />
</main>
`;
